src/System.Reflection/tests/FieldInfo/FieldInfo_ConstantTests.cs

```diff
@@ -11,67 +11,36 @@ namespace System.Reflection.Tests
 {
     public class FieldInfoConstantTests
     {
-        //Verify SetValue method does not set value for constant Fields
-        [Fact]
-        public void TestSetValue_constIntField()
-        {
-            TestSetValue_constantField("constIntField", (object)222);
-        }
-
-        //Verify SetValue method does not set value for constant Fields
-        [Fact]
-        public void TestSetValue_constStrField()
-        {
-            TestSetValue_constantField("constStrField", (object)"new value");
-        }
-
-        //Verify SetValue method does not set value for constant Fields
-        [Fact]
-        public void TestSetValue_constCharField()
+        public static IEnumerable<object[]> FieldInfoConstants_TestData()
         {
-            TestSetValue_constantField("charField", (object)'A');
+            yield return new object[] { "constIntField", 222 };
+            yield return new object[] { "constStrField", "new value" };
+            yield return new object[] { "charField", 'A' };
+            yield return new object[] { "boolField", false };
+            yield return new object[] { "floatField", 4.56 };
+            yield return new object[] { "doubleField", double.MaxValue };
+            yield return new object[] { "int64Field", long.MaxValue };
+            yield return new object[] { "byteField", byte.MaxValue };
```

Comment:
It looks like all of these could be done as InlineData rather than as MemberData.  InlineData is nice to use when possible, as it keeps the test data directly associated with the test method.  The value of MemberData is primarily that it allows the data to be reused across many tests and it allows for encoding of arguments that can't be encoded directly in attributes, but at first glance neither of benefits appear to apply here.